server/tests/test_notebook_view.py

```diff
@@ -149,6 +149,36 @@ def test_tryit_view(client, fake_user, logged_in, iomd):
         assert len(response.redirect_chain) == 0
 
 
+@pytest.mark.parametrize("logged_in", [True, False])
+@pytest.mark.parametrize("request_method", ["POST", "GET"])
+def test_tryit_view_file_uploads(client, fake_user, logged_in, request_method):
+    import tempfile
+    import mimetypes
+
+    # we should see the same script blocks
+    # for both logged_in and both request_method
+
+    with tempfile.NamedTemporaryFile(mode="w+") as f:
+        f.name = "data.txt"
+        file_content = "testing"
+        file_mimetype = mimetypes.guess_type(f.name)
+        f.write(file_content)
+        f.seek(0)
+        path = reverse("try-it")
+
+        def http_response(request_method):
+            if request_method == "POST":
```

Comment:
I didn't get into this earlier, but we won't be using `/tryit/` for `POST` requests-- for that, we'll be using a specialized endpoint (probably `/from-template/`). Let's create a seperate test as I think it'll be pretty different from this one.

This also eliminates the need to create a special helper function